packages/server-wallet/src/handlers/join-channel.ts

```diff
@@ -0,0 +1,62 @@
+import {Either, left, right, chain, map} from 'fp-ts/lib/Either';
+import {StateVariables} from '@statechannels/wallet-core';
+import {pipe} from 'fp-ts/lib/function';
+import {ChannelId} from '@statechannels/client-api-schema';
+
+import {SignState} from '../protocols/actions';
+import {ChannelState} from '../protocols/state';
+
+type HandlerResult = Either<JoinChannelError, SignState>;
+type StepResult = Either<JoinChannelError, ChannelState>;
+export interface JoinChannelHandlerParams {
+  channelId: ChannelId;
+}
+
+export enum Errors {
+  channelNotFound = 'channel not found',
+  invalidLatestState = 'latest state must be turn 0',
+  alreadySignedByMe = 'already signed prefund setup',
+}
+
+export class JoinChannelError extends Error {
+  readonly type = 'JoinChannelError';
+  constructor(reason: Errors, public readonly data: any = undefined) {
+    super(reason);
+  }
+}
+
+// The helper functions should be factored out, tested, and reusable
+const hasStateSignedByMe = (cs: ChannelState): boolean => !!cs.latestSignedByMe;
+
+const ensureNotSignedByMe = (cs: ChannelState): Either<JoinChannelError, ChannelState> =>
+  hasStateSignedByMe(cs) ? left(new JoinChannelError(Errors.alreadySignedByMe)) : right(cs);
+
+function ensureLatestStateIsPrefundSetup(cs: ChannelState): StepResult {
+  if (cs.latest.turnNum === 0) return right(cs);
+  return left(new JoinChannelError(Errors.invalidLatestState));
+}
```

Comment:
Seems to me like these could be defined in some a group to be more easily readable, like...

```ts
const joinChannelValidators = [ notSignedByMe, latestStateIsPrefundSetup ];
```

Reply:
I'm not sure if there's much added value in that, since the implementation of the `joinChannel` handler _is_ that group.

Do you think there's much more added value in doing something like
```
const joinChannelValidators = [ notSignedByMe, latestStateIsPrefundSetup ];
const joinChannelHandler = flow(...joinChannelValidators)
```
vs something like
```
const joinChannelHandler = flow(notSignedByMe, latestStateIsPrefundSetup)
```
?

Reply:
My immediate thought when reading both of these is "what is `flow`?".

Reply:
Do you mean `pipe`?

Reply:
https://gcanti.github.io/fp-ts/modules/function.ts.html#flow

I don't know why `flow` is not called `compose`...

Reply:
My comment was strictly written with code readability in mind... for the random passer-by of this file on Github, what small changes could we make to the way its written to maximize the likelihood they understand what's going on

Reply:
I'm not exactly sure what change you're proposing though. Are you suggesting something like the snippets above? Or just defining an unused constant `joinChannelValidators` for documentation purposes?